pypesto/sampling/auto_correlation.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+
+
+def auto_correlation(chain: np.array):
+    """
+    This function estimates the integrated autocorrelation time
+    using Sokal's adaptive truncated periodogram estimator.
+
+    Parameters
+    ----------
+    chain:
+        The MCMC parameter samples without warm up phase.
+        In case of sampling with parallel tempering, the
+        chain with the lowest temperature T=1 should be passed.
+
+    Returns
+    -------
+    tau:
+        Array with the auto-correlation time tau for each parameter
+        dimension. We suggest taking the maximum over all components.
+
+    """
+
+    nsimu, npar = chain.shape
+    tau = np.zeros((npar))
+
+    x = np.transpose(np.fft.fft(np.transpose(chain)))
```

Comment:
the FT is supposed to be taken over the nsimu axis. This you can specify via `axis=0`, then the two transposes should not be necessary.